bin/sensei.js

```diff
@@ -225,6 +226,29 @@ cli(
     })
     .demandCommand(1, 1, "One command must be specified")
     .strict()
+    .middleware([
+      function handleConfig(argv, yargs) {
+        if (!path.isAbsolute(argv.config)) {
+          argv.config = path.join(argv.material, argv.config);
+        }
+        const config = fs.existsSync(argv.config)
+          ? JSON.parse(fs.readFileSync(argv.config, "utf-8"))
+          : {};
+        for (k in argv) {
```

Comment:
I guess there may be a better stream-ish way to do this.

Reply:
```javascript
Object.keys(argv).reduce((c, k) => {if (!(k in yargs.parsed.defaulted && k in c)) {c[k] = argv[k]}; return c}, config)
```

Reply:
I think I would do it like this

```js
for (const [name, value] of Object.entries(argv)) {
  const isDefinedOnCli = !yargs.parsed.defaulted?.[name]
  const isDefinedInConfig = config[name] !== undefined
  if (isDefinedOnCli && !isDefinedInConfig) {
    config[name] = value
  }
})
```

Besides, shouldn't CLI options take precedence over config options ?

Reply:
I like the more explicit code.
But isn't it better to check a key exists than check corresponding value is undefined?

This existing code is doing the precedence of CLI over config: for every CLI arg, if it's not a default or it doesn't exists in config, then we put it in config (maybe overriding). Your proposed version is breaking this logic by using `&&` instead of `||`.

Reply:
> But isn't it better to check a key exists than check corresponding value is undefined?

It depends, but after a second though I think you're right that it makes more sense here to check using `in`.

> This existing code is doing the precedence of CLI over config: for every CLI arg, if it's not a default or it doesn't exists in config, then we put it in config (maybe overriding).

Right! I misread the code.

Reply:
Changed for explicit loop with explicit variables.